src/Pact/Gas.hs

```diff
@@ -95,5 +95,6 @@ freeGasEnv = GasEnv 0 0.0 (constGasModel 0)
 constGasModel :: Word64 -> GasModel
 constGasModel r = GasModel
   { gasModelName = "fixed " <> tShow r
+  , gasModelType = ConstantGasModel (fromIntegral r)
```

Comment:
Can you say more about how this addition relates to the decimal arithmetic work? It looks like you're adding new functionality that is unrelated to the decimal work per se.

Reply:
See: https://github.com/kadena-io/pact/pull/1217/files#diff-cab0577f6cd5a3953dabcc04e946f3c74429edb0f0a0dea8088cba1e4bdc7061R65

or, in code:
```haskell
getTransGasParams = do
  gUsed <- view eeGas >>= liftIO . readIORef
  gEnv <- view eeGasEnv
  let chargeFn = getCharge (view (geGasModel . to gasModelType) gEnv)
      gLim = view geGasLimit gEnv
  pure (Dec.TransGasParams (fromIntegral gUsed) (fromIntegral gLim) chargeFn)
  where
  getCharge (ConstantGasModel g) = const (chargeArithGas (fromIntegral g))
  getCharge _ = chargePactArith
 ```
 In other words: How do we "Disable" the gas charging function whenever we want to use a fixed gas charge of 0?  Turns out that `runGasModel` is insufficient anymore, it's a higher order function that's in terms of `Eval` which doesn't particularly help us at all when we're accounting for gas in a different monad (in this case, `Arith`).
 
 This is simply how I managed to port over "similar behavior gassing" to the `Arith` monad.